io/pci/pci_info_lscfg.py

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright: 2016 IBM
+# Author: Srimannarayana Murthy Maram <Maram.Srimannarayana.Murthy@ibm.com>
+
+"""
+PCI VPD Test.
+Needs to be run as root.
+"""
+
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import pci
+from avocado.utils import process
+import re
+
+
+class PciLsvpdInfo(Test):
+    '''
+    check lsvpd info
+    '''
+
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        smm = SoftwareManager()
+        for pkg in ["lsvpd", "pciutils"]:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("%s package is need to test" % pkg)
+        if process.system("vpdupdate", ignore_status=True, shell=True):
+            self.fail("CFG Update fails")
+
+    def test(self):
+        '''
+        Test
+        '''
+        error = []
+        for pci_addr in pci.get_pci_addresses():
+            self.log.info("Checking for PCI Address: %s\n\n", pci_addr)
+            cfg_output = pci.get_cfg(pci_addr)
+            self.log.info(cfg_output)
+            if cfg_output:
+
+                # Extracting pci ID from Description
+                if 'Description' in cfg_output:
+                    pci_id_ibm = re.search(r'[(]{1}\d+[)]{1}',
+                                           cfg_output['Description'])
+                    if pci_id_ibm:
+                        cfg_output['pci_id'] = pci_id_ibm.group()[1:-1]
+
+                # Slot Match
+                if 'YL' in cfg_output:
+                    sys_slot = pci.get_slot_from_sysfs(pci_addr)
+                    if sys_slot:
+                        sys_slot = sys_slot.strip('\0')
+                    cfg_slot = cfg_output['YL']
+                    self.log.info("Slot from sysfs: %s", sys_slot)
+                    self.log.info("Slot from lscfg: %s", cfg_slot)
+                    if sys_slot in [sys_slot, cfg_slot[:cfg_slot.rfind('-')]]:
+                        self.log.info("=======>>> slot matches perfectly\n\n")
+                    else:
+                        error.append(pci_addr + "-> slot")
+                        self.log.info("--->>Slot Numbers not Matched\n\n")
+                else:
+                    self.log.error("Slot info not available in cfg output\n")
+
+                # Device ID match
+                sys_pci_id_output = pci.get_pci_id_from_sysfs(pci_addr)
+                cfg_dev_id = cfg_output['pci_id'][4:]
+                sysfs_dev_id = sys_pci_id_output[5:-10]
+                sysfs_sdev_id = sys_pci_id_output[15:]
+                self.log.info("Device ID from sysfs: %s", sysfs_dev_id)
+                self.log.info("Sub Device ID from sysfs: %s", sysfs_sdev_id)
+                self.log.info("Device ID from cfg: %s", cfg_dev_id)
+                if cfg_dev_id == sysfs_sdev_id or cfg_dev_id == sysfs_dev_id:
+                    self.log.info("=======>>Device ID Match Success\n\n")
```

Comment:
Device ID matches

Reply:
@maramsmurthy As discussed please push the computing of device and sub_device_id to common code of get_cfg().
it will easy and clean code for verification part.